src/kibana/components/courier/data_source/_abstract.js

```diff
@@ -295,6 +295,23 @@ define(function (require) {
           };
 
           /**
+           * Evict Filters that doesn't match the index patterns
+           * @param {object} filter The filter to clean
+           * @returns {object}
+           */
+          var keepApplicableFilters = function (filter) {
+            if (!!(filter.index)) {
+              return filter.index === this.id;
+            } else if (!!(filter.meta) && !!(filter.meta.index)) {
+              return filter.meta.index === this.id;
+            } else if (!!(filter.geo_bounding_box)) {
+              var field = _.keys(filter.geo_bounding_box)[0];
```

Comment:
Yeah, I don't particularly like the behavior change that is taking place here, nor that we're checking for some specific filter type. Also, what is being done with that `field` variable?